src/main/java/ejercicio/banco/repository/CsvAccountRepository.java

```diff
@@ -0,0 +1,113 @@
+package main.java.ejercicio.banco.repository;
+
+import main.java.ejercicio.banco.dto.Account;
+import main.java.ejercicio.banco.util.CsvReader;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class CsvAccountRepository implements AccountRepository {
+
+    private CsvReader reader;
+
+    public CsvAccountRepository(){
+        this.reader = new CsvReader();
+    }
+
+    @Override
+    public List<Account> findAll(String filename) throws FileNotFoundException {
+
+
+        // Creates list of Accounts
+        List<Account> accounts = new ArrayList<Account>();
+
+        // Creates an user ID starting at 1, auto-incrementing every new user
+        int userId = 1;
+
+        List<String> in = reader.read(filename);
+
+        for (int x = 0; x<in.size(); x++) {
+            String line = in.get(x);
+            System.out.println(line);
+
+            // Assuming that the info is separated by "; " ,splits it
+            String[] parts = line.split(";");
+            String userName = parts[0];
+            float money = Float.valueOf(parts[1]);
+            String iban = parts[2];
+
+            //Prints user into console
+            System.out.println("User " + userId + " named " + userName + " has been processed");
+
+            // Creates a new account and adds it to the list
+            Account acc = new Account(userId, userName, money, iban);
+            accounts.add(acc);
+            // Increments Id counter
+            userId++;
+        }
+        return accounts;
+    }
+
+
+    @Override
+    public Account find(int id) throws FileNotFoundException {
+        //Since we dont want to give the method a file parameter, Im hardcoding the filename
+        String filename = String.join(File.separator, "src", "main", "resources", "csv", "accounts.csv");
+        List<String> in = reader.read(filename);
+
+        String line = in.get(id);
```

Comment:
Be careful here. You are assuming the users in the `accounts.csv` are sorted by ID, but this could not be like that in all the cases (imagine we remove two of them from the CSV).

You should iterate over the list and make sure the ID are exactly the same.